src/components/TokenBalance.tsx

```diff
@@ -77,16 +77,25 @@ function TokenBalance({
       : undefined
   const { decimalSeparator } = getNumberFormatSettings()
 
+  const hideHomeBalanceState = useSelector(hideHomeBalancesSelector)
+  const hideBalance = showHideHomeBalancesToggle && hideHomeBalanceState
   const balanceDisplay = hideBalance ? `XX${decimalSeparator}XX` : totalBalanceLocal?.toFormat(2)
 
-  if (tokenFetchError || tokenFetchLoading || tokensAreStale) {
-    // Show '-' if we haven't fetched the tokens yet or prices are stale
+  const RenderTotalTokenBalance = ({ balanceDisplay }: { balanceDisplay: string }) => {
```

Comment:
non blocking naming nit: I think we use render prefix if we call this as a function. Since its used as a component, you can drop the Render prefix